### PR TITLE
adapter: interport implementation

### PR DESCRIPTION
**Pull request by dev-dozer**

### CODE REVIEW

**Comment by xsteadybcgo** [comment]
@dev-dozer Thank you for submitting the PR; we are excited about it. Currently, the README does not clearly explain some data requirements, which we will further supplement. I have reviewed your subgraph code, and the logic does not yet meet our requirements. I have left comments in the corresponding parts of the code.

File: src/adapters/interport/execution/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { UserBalance } from './sdk/types';
+import {
+  getUserPositionsAtBlock,
+} from './sdk/lib';
+
+export const getUserBalanceByBlock = async (blockNumber: number, blockTimestamp: number): Promise<UserBalance[]> => {
```
Comment (xsteadybcgo) [inline src/adapters/interport/execution/src/index.ts:6]: The function signature has been modified; please use `function getUserTVLData(blockNumber: number)`.

File: src/adapters/interport/execution/src/sdk/utils/constants.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export const USDC_VAULT_ADDRESS = '0x22E7c5d443400eef48F58acdEC67986373cf6c74';
+export const USDT_VAULT_ADDRESS = '0xd9eF111006F7477aC6E70dD2B782D8A901f6Ff5b';
+export const STABLECOIN_FARM_ADDRESS = '0xEF902012C4a227461BdefF8Fa26b596eA125C711';
+
+export const RPC_URL = 'https://rpc.zklink.io';
+export const BACKUP_RPC_URL = 'https://rpc.zklink.network';
+
+export const SUBGRAPH_URL = 'http://3.114.68.110:8000/subgraphs/name/interport-stablecoinfarm-tracking-zklink-nova';
```
Comment (xsteadybcgo) [inline src/adapters/interport/execution/src/sdk/utils/constants.ts:8]: Set the `SUBGRAPH_URL` as an environment variable, referencing the **example** provided.

File: src/adapters/interport/execution/src/sdk/lib.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      userAddress: staking.user,
+      tokenAddress,
+      poolAddress: STABLECOIN_FARM_ADDRESS,
+      balance: BigInt(userBalance),
+      block_number: Number(staking.blocknumber),
+      timestamp: Number(staking.timestamp),
+    });
+  }
+
+  return results;
```
Comment (xsteadybcgo) [inline src/adapters/interport/execution/src/sdk/lib.ts:99]: The `poolAddress` should not be `STABLECOIN_FARM_ADDRESS`, but rather the address of the contract where assets are locked (which I understand should be iUSDC/iUSDT; we will verify which contract the user's assets are locked in and check whether that contract holds such assets). The `tokenAddress` here should not be the vault contract address (i.e., iUSDC/iUSDT), but should instead be the addresses of USDT and USDC.

File: src/adapters/interport/subgraph/subgraph.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      language: wasm/assemblyscript
+      file: ./src/mapping.ts
+      entities:
+        - UserStake
+      abis:
+        - name: StablecoinFarm
+          file: ./abis/StablecoinFarm.json
+      eventHandlers:
+        - event: Staked(indexed address,indexed uint256,uint256)
+          handler: handleStaked
```
Comment (xsteadybcgo) [inline src/adapters/interport/subgraph/subgraph.yaml:25]: The balance here is incorrect. By examining the output data, we found that the address [0xB68D6D933f7c14f6304927015De7b2021540ceC4](https://explorer.zklink.io/address/0xb68d6d933f7c14f6304927015de7b2021540cec4) does not have any user iUSDT and iUSDC assets. The user tvl data here refers to the proportion of assets locked by the user in the protocol relative to the total assets of the protocol. As mentioned in our documentation, for example, if users A and B have locked 50 USDC each, and the contract locks a total of 100 USDC at that time, then the balance of A and B would be 50. (If at this point A withdraws all the staked USDC, then A's balance would be 0; if at this point user C borrows 50 USDC, then the assets locked by the contract are only 50, and A and B each occupy 50%, thus A and B's balance would be 25 USDC each.)

Comment (xsteadybcgo) [reply src/adapters/interport/subgraph/subgraph.yaml:25]: I think you missed the withdraw event here; it should reduce the user's balance proportion.

File: src/adapters/interport/subgraph/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  "name": "zk_l3_graph",
+  "license": "UNLICENSED",
+  "scripts": {
+    "codegen": "graph codegen",
+    "build": "graph build",
+    "clean": "graph clean",
+    "create": "graph create interport-stablecoinfarm-tracking-zklink-nova --node http://3.114.68.110:8020",
+    "deploy": "graph deploy interport-stablecoinfarm-tracking-zklink-nova --ipfs http://3.114.68.110:5001 --node http://3.114.68.110:8020",
+    "remove": "graph remove interport-stablecoinfarm-tracking-zklink-nova --node http://3.114.68.110:8020",
+    "test": "graph test"
```
Comment (xsteadybcgo) [inline src/adapters/interport/subgraph/package.json:11]: Remove the scripts for `create`, `deploy`, and `remove`. The `interport-stablecoinfarm-tracking-zklink-nova` should not be exposed.

File: src/adapters/interport/execution/src/index.ts (hunk 1 above)
Comment (dev-dozer) [reply src/adapters/interport/execution/src/index.ts]: Done

File: src/adapters/interport/execution/src/sdk/utils/constants.ts (hunk 2 above)
Comment (dev-dozer) [reply src/adapters/interport/execution/src/sdk/utils/constants.ts]: Done

File: src/adapters/interport/subgraph/package.json (hunk 5 above)
Comment (dev-dozer) [reply src/adapters/interport/subgraph/package.json:8]: Done

**Pull request merged**